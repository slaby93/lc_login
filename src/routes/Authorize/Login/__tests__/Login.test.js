import React from 'react';
import { shallow } from 'enzyme';
import LoginStyled, { Login } from './../Login';

jest.mock('utils/form/canSubmit', () => true);

describe('<Login />', () => {
  test('CSS', () => {
    const mountedComponent = shallow(<LoginStyled
      onSubmit={jest.fn()}
    />);
    expect(mountedComponent).toMatchSnapshot();
  });

  test('HTML', () => {
    const mountedComponent = shallow(<Login
      onSubmit={jest.fn()}
      className="test"
    />);
    expect(mountedComponent).toMatchSnapshot();
  });
});

describe('errorValidator', () => {
  const { errorValidator } = Login;

  test('all valid', () => {
    expect(errorValidator({
      email: 'test@test.pl',
      password: 'Password1',
    })).toMatchSnapshot();
  });

  describe('email', () => {
    test('empty', () => {
      expect(errorValidator({
        email: '',
        password: 'Password1',
      })).toMatchSnapshot();
    });

    test('invalid format', () => {
      expect(errorValidator({
        email: 'test@',
        password: 'Password1',
      })).toMatchSnapshot();
    });
  });

  describe('password', () => {
    test('empty', () => {
      expect(errorValidator({
        email: 'test@test.pl',
        password: '',
      })).toMatchSnapshot();
    });

    test('lack of uppercase', () => {
      expect(errorValidator({
        email: 'test@test.pl',
        password: '1enough_chars',
      })).toMatchSnapshot();
    });

    test('lack of lowercase', () => {
      expect(errorValidator({
        email: 'test@test.pl',
        password: '1_ENOUGH_CHARS',
      })).toMatchSnapshot();
    });

    test('lack of number', () => {
      expect(errorValidator({
        email: 'test@test.pl',
        password: 'PasswordOne',
      })).toMatchSnapshot();
    });
  });
});
