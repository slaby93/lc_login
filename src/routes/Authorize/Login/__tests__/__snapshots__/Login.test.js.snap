// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> CSS 1`] = `
.c0 {
  min-width: 270px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 h1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-bottom: 60px;
}

.c0 .Checkbox__StyledComponent-bGxCTx {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<Login
  className="c0"
  isLogging={false}
  onSubmit={[Function]}
/>
`;

exports[`<Login /> HTML 1`] = `
<div
  className="test"
  data-automation-id="LoginForm"
>
  <Header__StyledComponent>
    Log In
  </Header__StyledComponent>
  <Form
    onSubmit={[Function]}
    validateError={[Function]}
  />
</div>
`;

exports[`errorValidator all valid 1`] = `
Object {
  "email": false,
  "password": false,
}
`;

exports[`errorValidator email empty 1`] = `
Object {
  "email": "Field is required",
  "password": false,
}
`;

exports[`errorValidator email invalid format 1`] = `
Object {
  "email": "Email has invalid format",
  "password": false,
}
`;

exports[`errorValidator password empty 1`] = `
Object {
  "email": false,
  "password": "Field is required",
}
`;

exports[`errorValidator password lack of lowercase 1`] = `
Object {
  "email": false,
  "password": "Provide lowercase character",
}
`;

exports[`errorValidator password lack of number 1`] = `
Object {
  "email": false,
  "password": "Provide number character",
}
`;

exports[`errorValidator password lack of uppercase 1`] = `
Object {
  "email": false,
  "password": "Provide uppercase character",
}
`;
